srcpkgs/yacy/template

```diff
@@ -0,0 +1,35 @@
+# Template file for 'yacy'
+pkgname=yacy
+version=1.922
+revision=1
+hostmakedepends="openjdk apache-ant"
+depends="openjdk-jre"
+short_desc="YaCy p2p decentralized search engine"
+maintainer="Jason Manley <jason@jasondavid.tv>"
+license="GPL-3.0-or-later"
+homepage="https://yacy.net"
+distfiles="http://latest.yacy.net/yacy_v${version}_20191013_9964.tar.gz"
+checksum=@99af963642dafaeab2657ad59315e7b7896bf4c1e7091f8f67e95e71802d44df
+
+system_accounts="yacy"
+yacy_homedir="/var/lib/yacy"
+yacy_shell="/bin/bash"
+make_dirs="/var/lib/yacy 0755 yacy yacy"
+
+post_extract() {
+	mv ${XBPS_BUILDDIR}/{yacy,yacy-1.922}
+}
+
+do_build() {
+	export JAVA_HOME=/usr/lib/jvm/java-1.8-openjdk
+	ant clean all
+}
+
+do_install() {
+	vmkdir var/lib/yacy
+	vcopy $XBPS_BUILDDIR/yacy-1.922/* var/lib/yacy
+}
```

Comment:
I'm not sure populating `/var` is the correct procedure. And you should just copy from inside the folder, no need to use `XBPS_BUILDDIR`.

Reply:
Put `pwd` inside this function and observe the build process if you want to know what directory you're in.

Reply:
I just pushed changes to condense this along the lines of what you just said.

Reply:
I've been trying several approaches, and am not sure which is the best here.

Reply:
So what I've noticed is everything works inside the ${wrksrc}, the compilation doesn't really put anything into ${DESTDIR}. Not sure how to proceed. I have a way that makes it work but /var/lib/yacy ends up owned by root:root instead of yacy:yacy .

Reply:
I also noticed in the documentation that there is no 'ant' or 'java' build style.

Reply:
Most of the templates using ant are copying files manually into destdir. You can use `vcopy` and `vinstall` from inside `do_install` for that.

Regarding the permissions of the folder, I'm not sure.

Reply:
The vcopy setup I have inside do_install() does work to get the files there. I tried some variants before that only copied DATA (which I think may have been the only subdir inside ${DESTDIR} from all that). For ownership, in what function might be best to try to set that? The homedirs directive has /var/lib/yacy with yacy as the owner, I wonder if anything I'm doing with do_install() is overwriting that?